File: frontend/property-pro-client/src/types/property.ts
export interface Property {
  propertyId: number
  name: string
  address: string
  city: string
  province: string
  postalCode: string
  propertyType: string
  createdAt: string
  totalUnits: number
  availableUnits: number
  occupiedUnits: number
}

export interface Unit {
  unitId: number
  propertyId: number
  propertyName: string
  unitNumber: string
  bedrooms: number
  bathrooms: number
  rentAmount: number
  squareFeet?: number
  status: 'Available' | 'Occupied' | 'UnderMaintenance'
  createdAt: string
}

export type PropertyType =
  | 'House'
  | 'Townhouse - Freehold'
  | 'Townhouse - Condo'
  | 'Condo Apartment'
  | 'Rental Apartment'
  | 'Multiplex'
  | 'Basement Apartment'
  | 'Room Rental'

export const PROPERTY_TYPES: PropertyType[] = [
  'House',
  'Townhouse - Freehold',
  'Townhouse - Condo',
  'Condo Apartment',
  'Rental Apartment',
  'Multiplex',
  'Basement Apartment',
  'Room Rental',
]

export const MULTI_UNIT_TYPES: PropertyType[] = [
  'Rental Apartment',
  'Multiplex',
  'Room Rental',
]

export const isMultiUnit = (type: string) =>
  MULTI_UNIT_TYPES.includes(type as PropertyType)

export type UnitStatus = 'Available' | 'Occupied' | 'UnderMaintenance'

File: backend/PropertyPro.API/Models/Property.cs
namespace PropertyPro.API.Models;

public class Property
{
    public int PropertyId { get; set; }
    public int LandlordId { get; set; }
    public User Landlord { get; set; } = null!;
    public string Name { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Province { get; set; } = string.Empty;
    public string PostalCode { get; set; } = string.Empty;
    public string PropertyType { get; set; } = string.Empty; // Apartment | House | Condo | Commercial
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Unit> Units { get; set; } = new List<Unit>();
    public ICollection<PropertyImage> Images { get; set; } = new List<PropertyImage>();
}
